laser_slam/include/laser_slam/laser_track.hpp

```diff
@@ -47,6 +47,9 @@ class LaserTrack {
   /// \brief Get one local cloud in world frame.
   void getLocalCloudInWorldFrame(const Time& timestamp, DataPoints* out_point_cloud) const;
 
+  /// \brief Get all laser scans.
+  const std::vector<LaserScan>* getLaserScans() const;
```

Comment:
Thanks for your input. 
No idea why `const std::vector<LaserScan>& laserScans = ... ;` feels wrong. It's just an intuition, which is apparently wrong :stuck_out_tongue: 
I like your second option. Would there be any problem with doing `for (auto&& scan : getLaserScans())` if `getLaserScans()` was to return a const ref?